docs/Developers/Design/CRD_design.md

```diff
@@ -0,0 +1,69 @@
+This document describes the set of [Custom Resource Definitions
+(CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+that will be used to deploy mixins for a storage system. The actual implementation
+of the resources described here will be phased in during development. The
+purpose of this document is to provide the overall structure, ensuring the end
+result provides necessary configurability in a user-friendly manner.
+
+# Overview
+
+A single Monstorak operator can take care of deployment of mixins for one or
+more storage systems. Each storage cluster should already be deployed within
+the kubernetes cluster, or the storage cluster could be running on hosts outside
+of kubernetes cluster. The capabilities of the operator remains same in both
+the modes of the deployment of storage cluster and the same set of CRDs should
+be used for both the cases.
+
+A given deployment configuration for storage systems contains a Custom Resource
+which contains configurations options like whether alerts are enabled, if
+dashboards are enabled, name of the storage system and version details of storage
+system.
+
+![Hierarchy of monstorak custom resources](crd_hierarchy.dot.png)
```

Comment:
There is no `Storage Monitoring Plan`.

Reply:
Ok, will remove this. I felt still we are using StorageMonitoringPlan CR.